File: 2022-2023/sourcecode/lab/lab12/InterfaceInjectBasicsJava/app/src/main/java/pl/edu/uwr/pum/interfaceinjectbasicsjava/di/ComputerComponent.java
package pl.edu.uwr.pum.interfaceinjectbasicsjava.di;

import dagger.Component;
import pl.edu.uwr.pum.interfaceinjectbasicsjava.MainActivity;
import pl.edu.uwr.pum.interfaceinjectbasicsjava.di.modules.AmdModule;

@Component(modules = AmdModule.class)
public interface ComputerComponent {
    void inject(MainActivity mainActivity);
}

File: 2022-2023/sourcecode/lab/lab12/InterfaceInjectiBasicsKotlin/app/src/main/java/pl/edu/uwr/pum/interfaceinjectibasicskotlin/di/ComputerComponent.kt
package pl.edu.uwr.pum.interfaceinjectibasicskotlin.di

import dagger.Component
import pl.edu.uwr.pum.interfaceinjectibasicskotlin.MainActivity
import pl.edu.uwr.pum.interfaceinjectibasicskotlin.di.modules.AmdModule

@Component(modules = [AmdModule::class])
interface ComputerComponent {
    fun inject(mainActivity: MainActivity)
}